dojo/db_migrations/0093_endpoint_host_migration.py

```diff
@@ -99,6 +103,44 @@ def clean_hosts(apps, schema_editor):
         raise FieldError('It is not possible to migrate database because there is/are {} broken endpoint(s). '
                          'Please check logs.'.format(len(broken_endpoints)))
 
+    to_be_deleted = set()
+    for product in Product_model.objects.all().distinct():
+        for endpoint in Endpoint_model.objects.all():
+            if endpoint.id not in to_be_deleted:
+
+                ep = endpoint_filter(
+                    protocol=endpoint.protocol,
+                    userinfo=endpoint.userinfo,
+                    host=endpoint.host,
+                    fqdn=endpoint.fqdn,
+                    port=endpoint.port,
+                    path=endpoint.path,
+                    query=endpoint.query,
+                    fragment=endpoint.fragment,
+                    product_id=product.pk
+                ).order_by('id')
+
+                if ep.count() > 1:
+                    ep_ids = [x.id for x in ep]
+                    to_be_deleted.update(ep_ids[1:])
+                    logger.debug("0: {}, 1+: {}, all: {}".format(ep_ids[0], ep_ids[1:], to_be_deleted))
+                    Endpoint_Status_model.objects\
+                        .filter(id__in=ep_ids[1:])\
+                        .update(endpoint=ep_ids[0])
+                    epss = Endpoint_Status_model.objects\
+                        .filter(endpoint=ep_ids[0])\
+                        .values('finding')\
+                        .annotate(total=Count('id'))\
+                        .filter(total__gt=1)
+                    for eps in epss:
+                        esm = Endpoint_Status_model.objects\
+                            .filter(finding=eps['finding'])\
+                            .order_by('-date')
+                        esm.exclude(id=esm[0].pk).delete()
```

Comment:
I think your approach is to keep the eps that was created latest? Might it be slightly better to keep the eps that was _updated_ the latest, i.e. order_by('-ast_modified') ?

Reply:
You are right, I pushed update